### PR TITLE
Fixed duplicates in connect dialog

### PR DESCRIPTION
**Pull request by theJenix**
Fixed duplicates in connect dialog - unable to test because I cant reproduce this at will

### CODE REVIEW

**Comment by reidbaker** [comment]
Doesn't look like it breaks everything :panda_face:

File: SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java [hunk 1]
```diff
@@ -296,19 +298,27 @@ private void onFindFinished(Intent intent) {
         
         //locally initiated device discovery...pop up a dialog for the user
         List<FoundGuest> guests = intent.getParcelableArrayListExtra(ConnectionService.EXTRA_GUESTS);
-        //pull out the known items, so we can preselect them
+        
+        //deduplicate and pull out the known items, so we can preselect them
+        Set<FoundGuest>  unique = new HashSet<FoundGuest>(guests);
+        List<FoundGuest> retained = new ArrayList<FoundGuest>();
```
Comment (ejohnson44) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:306]: What is the difference between retained and known?

Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:306]: retained are the items retained through deduplication...known are ones we've connected to before (so we can preselect them).

Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/components/NetworkFragment.java:306]: Retained is the list of items with no duplicates?

**Comment by reidbaker** [comment]
Looks good :dog2:

**Pull request merged**